modal/_utils/mount_utils.py

```diff
@@ -35,6 +35,21 @@ def validate_mount_points(
     return validated
 
 
+def validate_network_file_systems(network_file_systems: Mapping[Union[str, PurePosixPath], _NetworkFileSystem]):
+    if not isinstance(network_file_systems, dict):
+        raise InvalidError("network_file_systems must be a dict[str, NetworkFileSystem] where the keys are paths")
+    validated_network_file_systems = validate_mount_points("Network file system", network_file_systems)
+
+    for path, network_file_system in validated_network_file_systems:
+        if not isinstance(network_file_system, (_NetworkFileSystem)):
```

Comment:
Do we need to check both the private `_NetworkFileSystem` object and the synchronicity-wrapped `NetworkFileSystem` object?

Reply:
Hm, we don't do that for the volume validation, is that an error then?

Reply:
Yeah using the public object fails, but when experimenting with this locally and mounting a volume / file system they seem to always have the private object during validation. Is there a way to get around that?

Reply:
OK got it — thanks for checking.

That all sounds good vis-a-vis the type checks then. Do we have at least one test that validates this behavior through the public interface? (i.e. by setting up an app with a function decorator that uses the mount parameter improperly?) If we have that I think we're good to go!

Reply:
You mean like the test added in `test/network_file_system_test.py` or something different?